great_expectations/datasource/simple_sqlalchemy_datasource.py

```diff
@@ -10,7 +10,9 @@
 from great_expectations.datasource.new_datasource import BaseDatasource
 
 if TYPE_CHECKING:
-    from sqlalchemy.engine import Engine as SaEngine
+    from sqlalchemy.engine import (  # noqa: TID251 # type-checking only
+        Engine as SaEngine,
```

Comment:
This was a bit annoying. Would be nice if we could configure this rule to not flag `TYPE_CHECKING` only imports